src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -87,6 +105,39 @@ class JooqCalibrateDataRepository: CalibrateDataRepository
         return arrayList
     }
 
+    @Suppress("MagicNumber")
+    private fun getFilteredDataFromConnection(
+        conn: Connection,
+        filterQuery: FilterQuery): List<CalibrateResultRow> {
+        val resultSet: ResultSet
+        var query = FILTER_TEMPLATE
+
+        for ((field, values) in filterQuery) {
+            if (values.size > 0) {
+                val questionMarks = values.map { "?" }.joinToString(",")
+                query += "$field IN ($questionMarks) AND "
+            }
+        }
```

Comment:
I wonder about using jOOQ or Querydsl to build this instead of manually creating the SQL. Probably not a big deal as only building a small amount here. But it might be a bit nicer.

Reply:
so i tried installing exposed, said it needed java 17, then i was going to use jooq which ended up being a little bit of a headache so i searched online and found a pretty nice one called ktorm: https://www.ktorm.org/, I have gone with this one, it is quite nice but i thought the iterator definition was cool in the FilterQuery stuff and that will not work with this unfortuantely, either way I think this is cleaner overall albeit a bit more verbose